workflow-templates/command-compile.yml

```
@@ -0,0 +1,104 @@
+name: Compile Command
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+
+    # On pull requests and if the comment starts with `/compile`
+    if: github.event.issue.pull_request != '' && startsWith(github.event.comment.body, '/compile')
+
+    steps:
+      - name: Add reaction on start
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          token: ${{ secrets.COMMAND_BOT_PAT }}
+          repository: ${{ github.event.repository.full_name }}
+          comment-id: ${{ github.event.comment.id }}
+          reaction-type: "+1"
+
+      - name: Init arguments
+        # Init arguments
+        run: |
+          command="${{ github.event.comment.body }}"
+          arguments=($command)
+          echo "arg1=${arguments[1]}" >> "$GITHUB_ENV"
+          echo "arg2=${arguments[2]}" >> "$GITHUB_ENV"
+
+      - name: Init path
+        # Init path depending on which command is run
+        run: |    
+          if ${{ startsWith(env.arg1, '/') }}; then
+            echo "git_path=${{ github.workspace }}${{env.arg1}}" >> "$GITHUB_ENV"
+          else
+            echo "git_path=${{ github.workspace }}${{env.arg2}}" >> "$GITHUB_ENV"
+          fi
+
+      - name: Init branch
+        uses: xt0rted/pull-request-comment-branch@v1
+        id: comment-branch
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.COMMAND_BOT_PAT }}
+          fetch-depth: 0
+          ref: ${{ steps.comment-branch.outputs.head_ref }}
+      
+      - name: Read package.json node and npm engines version
+        uses: skjnldsv/read-package-engines-version-actions@v1
+        id: package-engines-versions
+        with:
+          fallbackNode: '^12'
+          fallbackNpm: '^6'
+
+      - name: Set up node ${{ steps.package-engines-versions.outputs.nodeVersion }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ steps.package-engines-versions.outputs.nodeVersion }}
+
```

```suggestion
              cache: npm

```
Add cache which might speed things up :) https://github.com/actions/setup-node/releases/tag/v2.2.0

---

We should do that everywhere then :wink:

---

Ah, but I don't know if that is any use with `npm ci` since it drops the node_modules every time :thinking:

---

A discussion on the cache for setup-node was opened here https://github.com/actions/setup-node/issues/304. With the ADR stating that it is the global npm cache that is cached founnd here https://github.com/actions/setup-node/blob/main/docs/adrs/0000-caching-dependencies.md :)